include/envoy/stats/scope.h

```diff
@@ -47,10 +48,21 @@ class Scope {
   virtual void deliverHistogramToSinks(const Histogram& histogram, uint64_t value) PURE;
 
   /**
+   * Creates a Counter from the stat name. Tag extraction will be performed on the name.
    * @param name The name of the stat, obtained from the SymbolTable.
    * @return a counter within the scope's namespace.
    */
-  virtual Counter& counterFromStatName(StatName name) PURE;
+  Counter& counterFromStatName(const StatName& name) {
+    return counterFromStatName(name, absl::nullopt);
+  }
+  /**
+   * Creates a Counter from the stat name and tags. If tags are not provided, tag extraction
+   * will be performed on the name.
+   * @param name The name of the stat, obtained from the SymbolTable.
+   * @param tags optionally specified tags.
+   * @return a counter within the scope's namespace.
+   */
+  virtual Counter& counterFromStatName(const StatName& name, StatNameTagVectorOptRef tags) PURE;
```

Comment:
can we rename this method so we don't have a virtual and non-virtual method with the same name?